tools/VscPreviewExtension/src/Processors/docfxPreviewProcessor.ts

```diff
@@ -0,0 +1,97 @@
+import { TempPreviewFileProcessor } from "../Common/tempPreviewFileProcessor";
+import { TempPreviewFileInformation } from "../Common/tempPreviewFileInformation";
+
+export class DocFXPreviewProcessor extends PreviewProcessor {
+    public navigationPort: string;
+    public isMarkdownFileChanged: boolean = false;
+    public markupResult: MarkupResult = new MarkupResult();
+
+    private _isFirstTime: boolean = false;
+    private _openPreviewPageCallback;
+    private _shouldWriteTempPreviewFile: boolean = false;
+    private _tempPreviewFileInformation: TempPreviewFileInformation;
+
+    constructor(context) {
+        super(context);
+    }
+
+    public startPreview(uri: Uri, callback) {
+        let docfxConfigFilePath = this.validEnvironment();
+        if (docfxConfigFilePath != undefined && docfxConfigFilePath != null) {
+            try {
+                this._tempPreviewFileInformation = TempPreviewFileProcessor.initializeTempFileInformation(PreviewProcessor.context, this.navigationPort, this.parseConfig(docfxConfigFilePath));
+            } catch (err) {
+                window.showErrorMessage(`[Environment Error]: ${err}`);
+            }
+            this._shouldWriteTempPreviewFile = true;
+            this._isFirstTime = true;
+            this._openPreviewPageCallback = callback;
+            this.updateContent(uri);
+        } else {
+            window.showErrorMessage(`[Exntension Error]: Please Open a DocFX project folder`);
+        }
+    }
+
+    private validEnvironment() {
+        let workspacePath = workspace.rootPath;
+        if (!workspacePath) {
+            return null;
+        } else {
+            let previewConfigFilePath = path.join(workspacePath, ConstVariables.previewConfigFileName);
+            let docfxConfigFileName = ConstVariables.docfxConfigFileName;
+            if (fs.existsSync(previewConfigFilePath)) {
+                let previewconfig = JSON.parse(fs.readFileSync(previewConfigFilePath).toString());
+                if (previewconfig.docfxConfigFileName != null) {
+                    docfxConfigFileName = previewconfig.docfxConfigFileName;
+                }
+            }
+            if (fs.existsSync(path.join(workspacePath, docfxConfigFileName))) {
+                return path.join(workspacePath, docfxConfigFileName);
+            } else {
+                return null;
+            }
+        }
+    }
+
+    protected appendTempPreviewFileInformation(request: ProxyRequest) {
+        if (this._shouldWriteTempPreviewFile) {
+            this._shouldWriteTempPreviewFile = false;
+            request.appendTempPreviewFileInformation(this._tempPreviewFileInformation);
+        }
+        return request;
+    }
+
+    private parseConfig(docfxConfigFilePath) {
+        // The first char maybe \uFEFF
+        let docfxConfig = JSON.parse(fs.readFileSync(docfxConfigFilePath).toString().replace(/^\uFEFF/, ''));
+        let outputFolder = "_site";
```

Comment:
? it is, this is a default value, if the config file includes the "dest" item, it will be the value of this item.

---
In reply to: [113856256](https://github.com/dotnet/docfx/pull/1612#discussion_r113856256) [](ancestors = 113856256)